Library/Homebrew/cask/tab.rb

```diff
@@ -0,0 +1,120 @@
+# typed: true
+# frozen_string_literal: true
+
+require "tab"
+
+module Cask
+  class Tab < ::AbstractTab
+    attr_accessor :uninstall_flight_blocks, :uninstall_artifacts
+
+    # Instantiates a {Tab} for a new installation of a cask.
+    def self.create(cask)
+      attributes = generic_attributes(cask).merge({
+        "tabfile"                 => cask.metadata_main_container_path/FILENAME,
+        "uninstall_flight_blocks" => cask.uninstall_flight_blocks?,
+        "runtime_dependencies"    => Tab.runtime_deps_hash(cask, cask.depends_on),
+        "source"                  => {
+          "path"         => cask.sourcefile_path.to_s,
+          "tap"          => cask.tap&.name,
+          "tap_git_head" => nil, # Filled in later if possible
+          "version"      => cask.version.to_s,
+        },
+        "uninstall_artifacts"     => cask.artifacts_list(uninstall_only: true),
+      })
+
+      # We can only get `tap_git_head` if the tap is installed locally
+      attributes["source"]["tap_git_head"] = cask.tap.git_head if cask.tap&.installed?
```

Comment:
These seems to be shared so might be worth pulling into a method if it's similar enough?